HW8_classes/classes.py

```diff
@@ -0,0 +1,193 @@
+import matplotlib.pyplot as plt
+
+# task 1
+
+
+class Dogs:
+    def __init__(self, age=0, weight=0, sex='undefined', breed='mongrel', color='mix'):
+        self.age = age
+        self.weight = weight
+        self.sex = sex
+        self.breed = breed
+        self.color = color
+        self.commands = set()
+
+    def grow(self, month=1):
+        self.age += month
+        self.weight += month*0.5
+
+    def train(self, *commands):
+        for command in commands:
+            self.commands.add(command)
+
+    def contest_allowed(self):
+        if self.age >= 12 and self.breed != 'mongrel' and len(self.commands) >= 3:
+            return True
+        else:
+            return False
+
+
+# task 2
+
+good_nucleotides = set('AGCUagcu')
+
+code = {"UUU": "F", "UUC": "F", "UUA": "L", "UUG": "L",
+        "UCU": "S", "UCC": "S", "UCA": "S", "UCG": "S",
+        "UAU": "Y", "UAC": "Y", "UAA": "*", "UAG": "*",
+        "UGU": "C", "UGC": "C", "UGA": "*", "UGG": "W",
+        "CUU": "L", "CUC": "L", "CUA": "L", "CUG": "L",
+        "CCU": "P", "CCC": "P", "CCA": "P", "CCG": "P",
+        "CAU": "H", "CAC": "H", "CAA": "Q", "CAG": "Q",
+        "CGU": "R", "CGC": "R", "CGA": "R", "CGG": "R",
+        "AUU": "I", "AUC": "I", "AUA": "I", "AUG": "M",
+        "ACU": "T", "ACC": "T", "ACA": "T", "ACG": "T",
+        "AAU": "N", "AAC": "N", "AAA": "K", "AAG": "K",
+        "AGU": "S", "AGC": "S", "AGA": "R", "AGG": "R",
+        "GUU": "V", "GUC": "V", "GUA": "V", "GUG": "V",
+        "GCU": "A", "GCC": "A", "GCA": "A", "GCG": "A",
+        "GAU": "D", "GAC": "D", "GAA": "E", "GAG": "E",
+        "GGU": "G", "GGC": "G", "GGA": "G", "GGG": "G"}
+
+RNA_to_DNA = {"A": "T", "a": "t", "U": "A", "u": "a",
+              "G": "C", "g": "c", "C": "G", "c": "g"}
+
+
+class RNA:
+    def __init__(self, sequence, protein_coding=True):
+        if type(sequence) != 'str' and set(sequence).issubset(good_nucleotides):
+            self.seq = sequence
+            self.protein_coding = protein_coding
+        else:
+            print('RNA should contain only nucleotides from set {AGCUagcu}')
+
+    def __repr__(self):
+        return self.seq
+
+    def translation(self):
+        if self.protein_coding is True:
+            codon = []
+            protein = []
+            for nucleotide in self.seq.upper():
+                codon.append(nucleotide)
+                if len(codon) == 3:
+                    protein.append(code[''.join(codon)])
+                    codon = []
+            print(''.join(protein))
+        else:
+            print('It is not a protein-coding RNA, translation is not possible')
+
+    def reverse_transcription(self):
+        dna = []
+        for nucleotide in self.seq:
+            dna.append(RNA_to_DNA[nucleotide])
```

Comment:
Если бы мы объявили словари в теле класса, то можно было бы писать вот так
```suggestion
            dna.append(self.RNA_to_DNA[nucleotide])
```